addons/account/models/account_payment.py

```diff
@@ -457,6 +473,190 @@ def unlink(self):
             raise UserError(_('It is not allowed to delete a payment that already created a journal entry since it would create a gap in the numbering. You should create the journal entry again and cancel it thanks to a regular revert.'))
         return super(account_payment, self).unlink()
 
+    @api.multi
+    def _prepare_payment_moves(self):
+        ''' Prepare the creation of journal entries (account.move) by creating a list of python dictionary to be passed
+        to the 'create' method.
+
+        Example 1: outbound with write-off:
+
+        Account             | Debit     | Credit
+        ---------------------------------------------------------
+        BANK                |   900.0   |
+        RECEIVABLE          |           |   1000.0
+        WRITE-OFF ACCOUNT   |   100.0   |
+
+        Example 2: internal transfer from BANK to CASH:
+
+        Account             | Debit     | Credit
+        ---------------------------------------------------------
+        BANK                |           |   1000.0
+        TRANSFER            |   1000.0  |
+        CASH                |   1000.0  |
+        TRANSFER            |           |   1000.0
+
+        :return: A list of Python dictionary to be passed to env['account.move'].create.
+        '''
+        all_move_vals = []
+        for payment in self:
+            company_currency = payment.company_id.currency_id
+
+            # Compute amounts.
+            if payment.payment_type in ('outbound', 'transfer'):
+                counterpart_amount = payment.amount
+                write_off_amount = payment.payment_difference_handling == 'reconcile' and -payment.payment_difference or 0.0
+                liquidity_line_account = payment.journal_id.default_debit_account_id
```

Comment:
Those two lines could be put outside the if/else

Reply:
No. `default_debit_account_id` in the `if`, `default_credit_account_id` in the `else`.